that_depends/providers/base.py

```diff
@@ -22,6 +24,9 @@ def __init__(self) -> None:
         super().__init__()
         self._override: typing.Any = None
 
+    def __deepcopy__(self, *_: object, **__: object) -> typing_extensions.Self:
```

Comment:
I think very good decision will be to left here comment with reference to issue and reference to litestar behaviour. Because at some point it may be not so easy to understand why this decision is been made

Reply:
good idea, thank you